### ISSUE TITLE
Nexus branch errors out on Youtube videos with 416

### ISSUE THREAD

**Issue by matejdro** [peak3d/inputstream.adaptive#1117]
### Describe the bug
Here is a clear and concise description of what the problem is:

Playing youtube videos with https://github.com/anxdpanic/plugin.video.youtube is broken on the Nexus branch, from https://github.com/xbmc/inputstream.adaptive/commit/6d6f3038b2bb286211c78c7d03660b8c27b4bf1a onward.

Problem can be reproduced on the https://jenkins.kodi.tv/blue/organizations/jenkins/xbmc%2Finputstream.adaptive/detail/Nexus/200/pipeline/ build, but not on the https://jenkins.kodi.tv/blue/organizations/jenkins/xbmc%2Finputstream.adaptive/detail/Nexus/199/pipeline/ build.

## Expected Behavior

Video should play

## Actual Behavior

Play fails with the `Download failed with error 416` error

### To Reproduce

1. Update inputstream.adaptive to at least commit 6d6f3038b2bb286211c78c7d03660b8c27b4bf1a
2. Attempt to play any Youtube video

### Debuglog

The debuglog can be found here:

[logs.txt](https://github.com/xbmc/inputstream.adaptive/files/10571088/logs.txt)

### Your Environment
Used Operating system:

 - [X] Android
 - [ ] iOS
 - [ ] tvOS
 - [ ] Linux
 - [ ] OSX
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: NVIDIA SHIELD Android TV with Android TV 11.0.0 API level 30, kernel: Linux ARM 32-bit version 4.9.141-tegra-gfad87a91fd8a
 - Kodi version: Kodi (20.0 (20.0.0) Git:20230115-389e701cb9)
 - inputstream.adaptive version: 6d6f3038b2bb286211c78c7d03660b8c27b4bf1a

*note: Once the issue is made we require you to update it with new information should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by CastagnaIT** [comment]
im not able to reproduce your problem
i tried play your video by using last commit in Nexus branch
on Windows and Android, on both the video is played correctly

make sure to install last nightly, artifacts that can be taken from here:
https://jenkins.kodi.tv/blue/organizations/jenkins/xbmc%2Finputstream.adaptive/detail/Nexus/204/artifacts/

**Comment by matejdro** [comment]
Installed nightly from your link, same issue.

Here is the full log from startup to (failed) playback:
[kodi.log.txt](https://github.com/xbmc/inputstream.adaptive/files/10571711/kodi.log.txt)

can you get anything from those logs?

**Comment by CastagnaIT** [comment]
this last log no, because debug info are disabled
the first one the only difference i can see is a different stream endpoint

```
your:
https://rr2---sn-5voxnp-hcml.googlevideo.com/videoplayback?expire=1675376583&ei=Z-PbY46MEoj-1gLJsrXwDw&ip=85.10.23.115&id=o-ANhKBNGCU1W6KR9qmpXZVaXuokQCVgCIlY4La2a_DF_r&itag=315&source=youtube&requiressl=yes&mh=J-&mm=31,29&mn=sn-5voxnp-hcml,sn-c0q7lnz7&ms=au,rdu&mv=m&mvi=2&pl=20&initcwndbps=857500&vprv=1&mime=video/webm&gir=yes&clen=1813641633&dur=907.850&lmt=1674456841609052&mt=1675354670&fvip=4&keepalive=yes&fexp=24007246&c=ANDROID&txp=5532434&sparams=expire,ei,ip,id,itag,source,requiressl,vprv,mime,gir,clen,dur,lmt&sig=AOq0QJ8wRQIgapLXcpMUCYb8Uy0Yu4Xe7Nfl7R4a-m0Vg-5Pvom25tkCIQDsygcLjSwJqTSCH23RZ91eO5-XeEnblh3BgC8qwfYb2Q==&lsparams=mh,mm,mn,ms,mv,mvi,pl,initcwndbps&lsig=AG3C_xAwRQIhAK8PiNmpwV6B8mL5VjYbchSRtK-ev_N7DR_gJBcdiVrHAiAdT7Qsd0Pb3yFk1B8340IJZhZzmWsamK8258L6U9XZKg==

mine:
https://rr1---sn-fpoq-ugos.googlevideo.com/videoplayback?expire=1675380964&ei=hPTbY_C8GpGLgAfOoZfQCg&ip=151.34.222.97&id=o-AMd-c84J0w9h3yT6UuMq93PeB-pxPcXDOh4zJvtqxFNp&itag=299&source=youtube&requiressl=yes&mh=J-&mm=31,29&mn=sn-fpoq-ugos,sn-hpa7znsz&ms=au,rdu&mv=m&mvi=1&pl=19&initcwndbps=985000&vprv=1&mime=video/mp4&gir=yes&clen=242810047&dur=907.849&lmt=1674445311128582&mt=1675358987&fvip=2&keepalive=yes&fexp=24007246&c=ANDROID&txp=5532434&sparams=expire,ei,ip,id,itag,source,requiressl,vprv,mime,gir,clen,dur,lmt&sig=AOq0QJ8wRQIgGT1gXbvYqCjZdCTUCB26vYhulfA7qZIVxNE0RmEeS8QCIQCDu093O4rPA3m7zyc5zKoijZPBq55GNPXwNIyPonaP8g==&lsparams=mh,mm,mn,ms,mv,mvi,pl,initcwndbps&lsig=AG3C_xAwRQIgOt2VyZi4ODT7IjUeTVn0cq94jRNVCkEtFukjd6tietcCIQDWKu-IXI2xL4bCju3FRK-x073oq_oTgTe9pKlhdcj82g==
```

but this depends on the YT service not InputStreamAdaptive

you could try attach a new log by enabling kodi debug, and, curl info by enabling component-specific logging
maybe i can try compare other info

**Comment by matejdro** [comment]
Ah sorry, my bad, forgot to turn on debug.

Here are two logs, debug + curl logging should be turned on. Opening Kodi and attempting to play the same video.

Nightly version from above link  - got 416:  [kodi.log-nightly.txt](https://github.com/xbmc/inputstream.adaptive/files/10572111/kodi.log-nightly.txt)


version d1c3776 (last version that still works) - video plays: [kodi.log-d1c3776.txt](https://github.com/xbmc/inputstream.adaptive/files/10572114/kodi.log-d1c3776.txt)

**Comment by matthuisman** [comment]
Could be 32bit android specific.  I've got issues with a certain stream as well that just won't play on Android 32bit on kodi 20. Fine on 19.5. And fine on android 64bit

Oh. Never-mind. Mine broken before this commit ij question

**Comment by matthuisman** [comment]
416 Range Not Satisfiable error response code indicates that a server cannot serve the requested ranges.

**Comment by CastagnaIT** [comment]
@matejdro you forgot to enable curl logging

**Comment by matejdro** [comment]
:man_facepalming: It looks like I had to explicitly confirm the dialog after enabling the cURL logging.

Well, third time's the charm, here is the new set of logs:

[kodi.log-d1c3776.txt](https://github.com/xbmc/inputstream.adaptive/files/10580410/kodi.log-d1c3776.txt)
[kodi.log-nightly.txt](https://github.com/xbmc/inputstream.adaptive/files/10580412/kodi.log-nightly.txt)

**Comment by glennguy** [comment]
Thanks @matejdro, will have a look at this further soon.

`2023-02-03 16:46:26.162 T:30816   debug <general>: Curl::Debug - HEADER_OUT: Range: bytes=220-0`

We're not calculating the right range for some reason...

**Comment by glennguy** [comment]
https://github.com/anxdpanic/plugin.video.youtube/blob/8960b925aa2b2255576dba84d667e0c95d9efe21/resources/lib/youtube_plugin/kodion/impl/xbmc/xbmc_items.py#L49

Breaking changes in https://github.com/xbmc/inputstream.adaptive/pull/1102 mean the above line needs to be altered.

Something like `list_item.setProperty('inputstream.adaptive.manifest_headers', play_item.get_headers())` should work, but for backwards compatibility perhaps both lines should be called.

Tagging @anxdpanic

**Comment by matthuisman** [comment]
hmm, i thought they meant to still be working:
https://github.com/xbmc/inputstream.adaptive/blob/Nexus/src/main.cpp#L73

**Comment by matthuisman** [comment]
also, looking at that addon source, it seems to be using stream_headers here:
https://github.com/anxdpanic/plugin.video.youtube/blob/master/resources/lib/youtube_plugin/kodion/impl/xbmc/xbmc_items.py#L56 which will be used as manifest_headers for backwards compat

i also cant see the addon setting a range header either - so that must be IA?
And you said its calculated wrong?

**Comment by glennguy** [comment]
My bad, I forget details of these conversations sometimes, @anxdpanic please disregard unless further investigation. While you're here though the plan is to remove support for manifest headers using the pipe method in Kodi 21.

> so that must be IA?

For segment downloads that's correct, it's done in AdaptiveStream.cpp. Maybe there's an issue in the generated mpd.

@matejdro could you please provide a copy of the generated .mpd file?
On Windows it's stored in cache/plugin.video.youtube under the kodi folder, not sure about Android.
If you disable component logging the debug log will be much easier to read. Shortly after starting the stream you should be able to see the filename i.e `2023-02-05 16:42:50.689 T:10336   debug <general>: AddOnLog: inputstream.adaptive: Download finished: http://127.0.0.1:50152/WhaTru5czpA.mpd (downloaded 8600 byte, speed 23831267.19 byte/s)`, then have a look for `WhaTru5czpA.mpd` in the folder mentioned above.

**Comment by anxdpanic** [comment]
Thanks for the ping. I'll get it updated now while I'm at it 👍

**Comment by anxdpanic** [comment]
@matejdro https://github.com/anxdpanic/plugin.video.youtube/releases/download/6.x.x-dev/plugin.video.youtube-6.8.25.alpha1+matrix.1.zip if you want to try with the changes that were mentioned in this thread.

**Comment by pbasista** [comment]
The same HTTP error 416 occurs on armv7l architecture (32 bit ARM, Raspberry Pi 3 Model B) with Kodi 20.0 and InputStream Adaptive 20.3.2.

The issue is that the HTTP header `Range` appears to contain an invalid range definition. According to the [specification](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Range), the range start should be the first parameter but InputStream Adaptive seems to use the range end as the first parameter:

```log
Curl::Debug - HEADER_OUT: Range: bytes=228-0
```

**Comment by CastagnaIT** [comment]
i found a couple of regressions but i dont know if full fix this problem
you can test the fixes from the test builds:
https://jenkins.kodi.tv/blue/organizations/jenkins/xbmc%2Finputstream.adaptive/detail/PR-1119/1/artifacts

**Comment by matejdro** [comment]
Updated youtube version linked above does not change anything. Here are logs from it: 
[kodi-nightlyInputStream-youtube6.8.25.log](https://github.com/xbmc/inputstream.adaptive/files/10664244/kodi-nightlyInputStream-youtube6.8.25.log)

However, using new inputstream.adaptive from PR 1199 resolves the issue (even with regular 6.8.24 Youtube release).

Do you still need .mpd files from me?

**Comment by CastagnaIT** [comment]
if PR1199 for you works correctly we have solved the problem
so no need other data thanks
